src/main/java/org/brapi/test/BrAPITestServer/service/SearchQueryBuilder.java

```diff
@@ -16,23 +12,39 @@
 public class SearchQueryBuilder<T> {
 
 	private String selectClause;
+	private String selectOnlyIds;
 	private String whereClause;
 	private String sortClause;
 	private Map<String, Object> params;
 	private Class<T> clazz;
 
 	public SearchQueryBuilder(Class<T> clazz) {
 		this.selectClause = "SELECT distinct entity FROM " + clazz.getSimpleName() + " entity ";
+		this.selectOnlyIds = "SELECT entity.id FROM " + clazz.getSimpleName() + " entity ";
 		this.whereClause = "WHERE 1=1 ";
 		this.sortClause = "";
 		this.params = new HashMap<>();
 		this.clazz = clazz;
 	}
 
 	public String getQuery() {
+		if (sortClause.isEmpty()) {
+			// By default, sort on entity id to have query result remain idempotent
+			sortClause = " ORDER BY entity.id ASC ";
+		}
+
 		return selectClause + whereClause + sortClause;
 	}
 
+	public String getIdQuery() {
+		if (sortClause.isEmpty()) {
+			// By default, sort on entity id to have query result remain idempotent
+			sortClause = " ORDER BY entity.id ASC ";
+		}
```

Comment:
this could be moved to the constructor, line 25. Just to avoid maintaining the default in multiple places